src/cmap/connection.ts

```diff
@@ -369,7 +372,28 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
 
     // always emit the message, in case we are streaming
     this.emit('message', message);
-    const operationDescription = this[kQueue].get(message.responseTo);
+    let operationDescription = this[kQueue].get(message.responseTo);
+
+    if (!operationDescription && this.isMonitoringConnection) {
+      // This is how we recover when the initial hello's requestId is not
+      // the responseTo when hello responses have been skipped:
+
+      // First check if the map is of invalid size
+      if (this[kQueue].size > 1) {
+        this.onError(new MongoRuntimeError(INVALID_QUEUE_SIZE));
+      } else {
+        // Get the first orphaned operation description.
+        const entry = this[kQueue].entries().next();
+        if (entry) {
```

Comment:
Under what conditions could this evaluate to false?  Or is it defensive programming?

Reply:
It's defensive as Typescript will fail to compile since the call to `next()` can be undefined. This should never happen.

Reply:
Should an error be raised instead, or is this the normal way of handling such situations?

Reply:
We can't error handle here as if there was in fact no entry then we have no callback to pass the error to as it hangs off the operation description, so we fall back to just returning below. This is obviously not ideal but we have a ticket for that (NODE-4820) to refactor the queue out entirely.